res_users_password_security/models/res_users.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from datetime import datetime, timedelta
+import re
+
+from openerp import api, fields, models, _
+from openerp.addons.res_users_password_security.exceptions import PassError
+
+
+def delta_now(**kwargs):
+    dt = datetime.now() + timedelta(**kwargs)
+    return fields.Datetime.to_string(dt)
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+    password_write_date = fields.Datetime(
+        'Latest password update', readonly=True
+    )
+
+    @api.model
+    def create(self, vals):
+        vals['password_write_date'] = fields.Datetime.now()
+        return super(ResUsers, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        if vals.get('password'):
+            self.check_password(vals['password'])
+            vals['password_write_date'] = fields.Datetime.now()
+        return super(ResUsers, self).write(vals)
+
+    @api.multi
+    def password_match_message(self):
+        company_id = self.company_id
+        message = [
+            'Password must be %d characters or more.' %
+            company_id.password_length
+        ]
+        message.append('Must contain the following:')
```

Comment:
What if all checkboxes are set to false? you would still get the message while not necessary, right?